cxxheaderparser/parser.py

```diff
@@ -2374,20 +2383,40 @@ def _parse_type(
 
             tok = get_token()
 
-        if pqname is None:
-            if not pqname_optional:
-                raise self._parse_error(tok)
-            parsed_type = None
-        else:
-            # Construct a type from the parsed name
-            parsed_type = Type(pqname, const, volatile)
+        # Construct a type from the parsed name
+        parsed_type = Type(pqname, const, volatile)
 
         self.lex.return_token(tok)
 
         # Always return the modifiers
         mods = ParsedTypeModifiers(vars, both, meths)
         return parsed_type, mods
 
+    def _parse_member_operator(self) -> Operator:
+        """This function parses operator from class body."""
+        ctype, cmods = self._parse_type(None)
+        if not isinstance(ctype, Type):
+            raise self._parse_error(None)
+        pqname = PQName([NameSpecifier("operator")])
+        return Operator(pqname, "conversion", ctype, cmods)
+
+    def _parse_free_operator(
+        self,
+        pqname: PQName,
+        op: str,
+        mods: ParsedTypeModifiers,
+        const: bool,
+        volatile: bool,
+    ) -> Operator:
+        """This function parses operator implemented outside class body."""
+        last_seg = pqname.segments[-1]
+        assert last_seg.name.startswith("operator")
```

Comment:
The operator name gets concatenated in `_parse_pqname_name`... I think it would be better for that function to not concatenate and let the eventual user of the operator make that call.

Reply:
ok, I changed `_parse_pqname_name` to not concatenate this. This caused even more `large change for something that seems like it should be rather simple`, because I changed return types for several functions from `str|None` to `list[LexToken]`.